### DISCUSSION TITLE
How to prevent V2Ray DNS leak?

### DISCUSSION THREAD

**Discussion by andruska** [v2fly/v2ray-core#2534] [General]
[IPleak.net](https://ipleak.net/) shows the DNS servers set on both the V2Ray client (desktop) and the V2Ray server (VPS). 60% of DNS requests go through the servers specified in the desktop, 40% through the servers specified in the VPS. When V2Ray is turned off, all requests go 100% through the servers specified on the desktop. Desktop uses systemd-resolved.

Client configuration:

```
{
    "dns": {
        "servers": [
            "127.0.0.53"
        ]
    },
    "outbounds": [
        {
            "_QV2RAY_USE_GLOBAL_FORWARD_PROXY_": false,
            "mux": {
                "concurrency": 1,
                "enabled": null
            },
            "protocol": "vmess",
            "sendThrough": "0.0.0.0",
            "settings": {
                "vnext": [
                    {
                        "address": "xxx-xxx-xxx-xxx",
                        "port": 18686,
                        "users": [
                            {
                                "id": "xxx-xxx-xxx-xxx"
                            }
                        ]
                    }
                ]
            },
            "streamSettings": {
                "network": "ws",
                "wsSettings": {
                    "path": "/wspath"
                }
            },
            "tag": "xxxxxx"
        }
    ]
}

```

**Comment by tess-lu** [comment]
There is a [YouTube tutorial](https://www.youtube.com/watch?v=fqREM6b25SY) about this. It's in Chinese but you could turn on subtitle translation to understand it.

**Comment by robberphex** [comment]
Some possible point:
* tcp dns query
* ipv6 dns query
* and how do you proxy dns traffic
